store_flakes

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import ssl
+import sqlite3
+import urllib.request
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Store information about flaky tests')
+    parser.add_argument("--no-check-certificate", action="store_true", help="Don't validate the server's certificate")
+    parser.add_argument("--db", default="cockpit-flakes.db", help="Database name")
+    parser.add_argument('log', help="File log file")
+    opts = parser.parse_args()
+
+    db_conn = sqlite3.connect(opts.db)
+    cursor = db_conn.cursor()
+
+    init_db(cursor)
+
+    ctx = ssl.create_default_context()
+    if opts.no_check_certificate:
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+
+    if opts.log.startswith("http"):
+        fp = urllib.request.urlopen(opts.log, context=ctx)
+    else:
+        fp = open(opts.log, "r")
+
+    for line in find_failed_tests(fp):
+        (test, name, retry) = process_line(line)
+        insert(cursor, test, name, 1 if retry else 0)
+
+    fp.close()
+
+    db_conn.commit()
+    db_conn.close()
+
+
+def init_db(cursor):
+    cursor.execute("CREATE TABLE if not exists CockpitFlakes (test text, name text, retry integer)")
+
+
+def insert(cursor, test, name, retry):
+    cursor.execute("INSERT INTO CockpitFlakes VALUES ('{0}', '{1}', '{2}')".format(test, name, retry))
+
+
+def find_failed_tests(fp):
+    for line in fp:
+        if isinstance(line, bytes):
+            line = line.decode('utf-8')
+        if line.startswith("not ok "):
+            yield line
+
+
+def process_line(line):
+    fields = line.split(" ")
+    return (fields[3], fields[4], "RETRY" in line)
```

Comment:
This is underdocumented and sounds overly general -- let's please move the splitting into `find_failed_tests()` and directly return the relevant information as a tuple.